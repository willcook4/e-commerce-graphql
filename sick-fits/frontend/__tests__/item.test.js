
import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: 'ABCD123',
  title: 'A cool item',
  price: 4000,
  description: 'This is a really cool item',
  image: 'dog.jpg',
  largeImage: 'largedog.jpg'
}

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
      const wrapper = shallow(<ItemComponent item={fakeItem} />)
      expect(toJSON(wrapper)).toMatchSnapshot()
    })
})

/** 
 *  SHALLOW TESTING
 * 
 * 
describe('<Item/>', () => {
  it('renders the image properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />)
    // console.log(wrapper.debug()) 
    const img = wrapper.find('img')
    // console.log(img.props())
    expect(img.props().src).toBe(fakeItem.image)
    expect(img.props().alt).toBe(fakeItem.title)
  })
  
  it('renders the price tag and title properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />)
    // console.log(wrapper.debug())    
    const priceTag = wrapper.find('PriceTag')
    // console.log(PriceTag.debug())
    expect(priceTag.children().text()).toBe('$50')

    const title = wrapper.find('Title a')
    // console.log(Title.debug())
    expect(title.text()).toBe(fakeItem.title)
  })

  it('renders out the buttons properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />)
    // console.log(wrapper.debug())
    const buttonList = wrapper.find('.buttonList')
    console.log(buttonList.debug())
    expect(buttonList.children()).toHaveLength(3)
    // console.log(buttonList.debug())
    expect(buttonList.find('Link')).toHaveLength(1)
    expect(buttonList.find('Link').exists()).toBe(true)

    expect(buttonList.find('AddToCart').exists()).toBe(true)
    expect(buttonList.find('DeleteItem').exists()).toBe(true)
  })
})

*/