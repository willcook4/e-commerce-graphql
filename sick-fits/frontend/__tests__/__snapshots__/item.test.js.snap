// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABCD123",
          },
        }
      }
    >
      <a>
        A cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40
  </PriceTag>
  <p>
    This is a really cool item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABCD123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="ABCD123"
    />
    <DeleteItem
      id="ABCD123"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
